include/cuda/detail/__cuda_util

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX_CUDA_UTIL
+#define _LIBCUDACXX_CUDA_UTIL
+
+#include "cuda/std/detail/__config"
+#include "cuda/std/detail/__pragma_push"
+
+#include <iosfwd>  // for string forward decl
+
+#include <cuda/exception>
+
+#ifdef _LIBCUDACXX_NO_EXCEPTIONS
+#include <cstdlib>
+#endif
+#include "cuda/std/detail/__pragma_push"
+
+#include <cuda_runtime_api.h>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+namespace detail {
+inline _LIBCUDACXX_INLINE_VISIBILITY
+void __throw_on_cuda_error(::cudaError_t __error, char const *__msg) {
+  if (::cudaSuccess != __error) {
+    ::cudaGetLastError(); // Clear CUDA error state
+#ifndef _LIBCUDACXX_NO_EXCEPTIONS
+    throw cuda::cuda_error{__error, __msg};
+#else
+    ::std::abort();
+#endif
+  }
+}
+
+void __throw_on_cuda_error(::cudaError_t __error) {
+  __throw_on_cuda_error(__error, "");
+}
```

Comment:
This is missing inline, which causes linker errors when I include `<cuda/stream_view>` from multiple TUs linked together.

It's understandable why current tests wouldn't catch this. Perhaps there should be a test that compiles two TUs that each `#include` all libcu++ headers and attempts to link them together into a single binary?

```suggestion
inline _LIBCUDACXX_INLINE_VISIBILITY
void __throw_on_cuda_error(::cudaError_t __error) {
  __throw_on_cuda_error(__error, "");
}
```

Reply:
Fixed.

> Perhaps there should be a test that compiles two TUs that each #include all libcu++ headers and attempts to link them together into a single binary?

That's probably a good idea, though I'd say it's be out of scope of this PR. It's already stretching separation of concerns to the limit and probably a quite a bit beyond.

Reply:
Yeah I meant for libcu++, not this PR.